divi/src/wallet.cpp

```diff
@@ -180,35 +178,22 @@ bool WriteTxToDisk(const CWallet* walletPtr, const CWalletTx& transactionToWrite
     return CWalletDB(walletPtr->strWalletFile).WriteTx(transactionToWrite.GetHash(),transactionToWrite);
 }
 
-CWallet::CWallet(
+CWallet::CWallet(const CChain& chain, const BlockMap& blockMap
     ): cs_wallet()
-    , fFileBacked()
-    , strWalletFile()
     , transactionRecord_(new WalletTransactionRecord(cs_wallet,strWalletFile) )
     , outputTracker_( new SpentOutputTracker(*transactionRecord_) )
+    , chainActive_(chain)
+    , mapBlockIndex_(blockMap)
     , orderedTransactionIndex()
-    , nWalletVersion()
-    , fBackupMints()
-    , mapKeyMetadata()
-    , mapMasterKeys()
-    , nMasterKeyMaxID()
-    , mapAddressBook()
-    , vchDefaultKey()
-    , setLockedCoins()
-    , mapHdPubKeys()
-    , nNextResend()
-    , nLastResend()
     , pwalletdbEncryption(NULL)
-    , setInternalKeyPool()
-    , setExternalKeyPool()
-    , walletStakingOnly()
```

Comment:
Why the removal of the explicit initializations of the member variables?

Reply:
Because that code is redundant, as they will be default-constructed in any case.  I feel that it is much nicer to read in the code if you omit those redundant calls and just focus on the constructors that are *not default*.

But if you prefer to have the explicit calls for all member variables here, I can of course put them back.

Reply:
Yeah, definitely prefer all the member types to be explicitly constructed as opposed to letting the compiler decide, since it may choose to do weird things when it runs with optimization flags.

Reply:
In this case, though, the C++ standard specifies that the code has the same effect.  There is no undefined behaviour here or other weird things.  See, for instance, https://www.learncpp.com/cpp-tutorial/constructors/, "classes containing classes".

So the reason to prefer one over the other is purely for code readability.  I prefer to not list those that are default, but if you want to have them there, we can of course do it.

Reply:
Ah, I noticed only now that some of the variables are actually primitive types and not objects (for objects, the C++ standard guarantees that they will be default constructed even if not explicitly mentioned in the initialiser list).

I personally would never write an initialiser for a primitive type without specifying the "zero" value explicitly, as that seems confusing to me.  That's why I assumed all of them were objects without looking closer.

I've now updated the PR to keep those initialisers that are primitive in the list, and have an explicit value as well.  I kept removal of those that are objects, as I think it is easier to read the code and clearer what the intent is if you don't explicitly specify the default constructed ones.  But let me know if you want me to go back to the exact old version of course.